internal/transport/grpchttp2/http2bridge.go

```diff
@@ -0,0 +1,219 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpchttp2
+
+import (
+	"io"
+
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/hpack"
+	"google.golang.org/grpc/mem"
+)
+
+// FramerBridge is a struct that works as an adapter for the net/x/http2
+// Framer implementation to be able to work with the grpchttp2.Framer interface.
+// This type exists to give an opt-out feature for the new framer and it is
+// eventually going to be removed.
+type FramerBridge struct {
+	framer *http2.Framer  // the underlying http2.Framer implementation to perform reads and writes.
+	pool   mem.BufferPool // a pool to reuse buffers when reading.
+}
+
+// NewFramerBridge creates a new framer by taking a writer and a reader,
+// alongside the maxHeaderListSize for the maximum size of the headers table.
+// The underlying framer uses the SetReuseFrames feature to avoid extra
+// allocations.
+func NewFramerBridge(w io.Writer, r io.Reader, maxHeaderListSize uint32) *FramerBridge {
+	fr := &FramerBridge{
+		framer: http2.NewFramer(w, r),
+		pool:   mem.DefaultBufferPool(),
+	}
+
+	fr.framer.SetReuseFrames()
+	fr.framer.MaxHeaderListSize = maxHeaderListSize
+	fr.framer.ReadMetaHeaders = hpack.NewDecoder(initHeaderTableSize, nil)
+
+	return fr
```

Comment:
But once I typed the above comment, I had a thought as to whether it makes sense to accept the pool as a parameter instead of defaulting to the default pool.

Totally totally optional is the following: go/go-style/best-practices#option-structure. Even with the pool as a parameter, you will only have 4 arguments. So, not a big deal at this point.